stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api/models/ProgressRequestBody.kt

```diff
@@ -25,20 +29,43 @@ internal class ProgressRequestBody(
         val total = file.length()
         val buffer = ByteArray(DEFAULT_BUFFER_SIZE)
         var uploaded = 0L
+
         FileInputStream(file).use { fis ->
             var read: Int
-            val handler = Handler(Looper.getMainLooper())
             while (fis.read(buffer).also { read = it } != -1) {
-                handler.post {
-                    callback.onProgress((100 * uploaded / total))
-                }
-                uploaded += read.toLong()
                 sink.write(buffer, 0, read)
+                uploaded += read.toLong()
+                withCallback { onProgress((100 * uploaded / total)) }
```

Comment:
We still have an issue here which is that we're just writing into a buffer, and showing the write progress of that operation on the UI. This means we get to things like `10 MB / 10 MB` rather quickly but then keep spinning because the network upload is actually still happening (as far as I understand).

There are mentions of this issue in a bunch of places, but I didn't find a good solution so far:
- https://github.com/square/retrofit/issues/431#issuecomment-111259238
- Comments below https://stackoverflow.com/a/44125313/4465208